stacks-signer/src/v0/signer.rs

```diff
@@ -567,33 +613,127 @@ impl Signer {
             .unwrap_or_else(|_| panic!("FATAL: total weight exceeds u32::MAX"))
     }
 
+    /// Handle an observed rejection from another signer
+    fn handle_block_rejection(&mut self, rejection: &BlockRejection) {
+        debug!("{self}: Received a block-reject signature: {rejection:?}");
+
+        let block_hash = &rejection.signer_signature_hash;
+        let signature = &rejection.signature;
+
+        let mut block_info = match self.signer_db.block_lookup(self.reward_cycle, block_hash) {
+            Ok(Some(block_info)) => {
+                if block_info.state == BlockState::GloballyRejected
+                    || block_info.state == BlockState::GloballyAccepted
+                {
+                    debug!("{self}: Received block rejection for a block that is already marked as {}. Ignoring...", block_info.state);
+                    return;
+                }
+                block_info
+            }
+            Ok(None) => {
```

Comment:
It's possible that the signer just hasn't seen this block yet.  What would be the consequence of dropping legit rejection messages that happen to race the block proposal and win?

Reply:
I debated adding a todo here about this actually because it definitely does add a race condition between one signer seeing a block proposal or not. Its a bit involved to fix as the rejection message does not include the block proposal itself. If it did, we could just store it and then start processing the block. Otherwise, we could store these orphaned responses and attempt to apply them following a block proposal's arrival. If you think this is worthwhile to handle, I would open a ticket rather than address in this PR.

Reply:
Yeah, let's open a ticket.  It might not even be worth addressing, because a block that is globally rejected will never get built on anyway, even if this signer doesn't see its rejection.

Reply:
Done https://github.com/stacks-network/stacks-core/issues/5146